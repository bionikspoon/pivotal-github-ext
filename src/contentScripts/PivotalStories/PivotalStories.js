/* @flow */
import React from 'react'
import { map } from 'ramda'
import PivotalStory from '../PivotalStory'
import './PivotalStories.css'

type Props = {
  storyIds: Array<string>,
}
export default function PivotalStories(props: Props) {
  return (
    <div className="PivotalStories__container">
      {map(
        storyId => <PivotalStory storyId={storyId} key={storyId} />,
        props.storyIds
      )}
    </div>
  )
}
