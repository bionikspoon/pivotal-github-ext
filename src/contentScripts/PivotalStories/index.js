/* @flow */
export { default } from './PivotalStories'
