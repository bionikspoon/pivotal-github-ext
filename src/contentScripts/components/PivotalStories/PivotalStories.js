/* @flow */
import React from 'react'
import { map } from 'ramda'
import PivotalStory from '../PivotalStory'
import './PivotalStories.css'

type Props = {
  storyIds: Array<string>,
}

export default function PivotalStories(props: Props) {
  const renderStories = map(storyId => (
    <PivotalStory storyId={storyId} key={storyId} />
  ))
  return (
    <div className="PivotalStories__container">
      {renderStories(props.storyIds)}
    </div>
  )
}
