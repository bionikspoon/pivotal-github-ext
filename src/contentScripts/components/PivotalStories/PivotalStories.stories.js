import React from 'react'

import { storiesOf } from '@storybook/react'
import PivotalStories from './'
import GithubWrapper from '../GithubWrapper'

storiesOf('PivotalStories', module)
  .addDecorator(story => <GithubWrapper>{story()}</GithubWrapper>)
  .add('default', () => (
    <PivotalStories
      storyIds={['151356728']}
      storage={storage}
      fetchResources={fetchResources}
    />
  ))

async function storage(...args) {
  console.log('storage ...args', ...args)
  return { pivotalApiKey: '123' }
}

async function fetchResources(...args) {
  console.log('fetchResources ...args', ...args)
  return {
    story: {
      kind: 'story',
      id: 151356728,
      created_at: '2017-09-24T21:05:03Z',
      updated_at: '2017-12-17T19:47:48Z',
      estimate: 1,
      story_type: 'feature',
      name: 'Test Story',
      description:
        'Really need this story to test things\n\n```\nI like turtles\n```',
      current_state: 'started',
      requested_by_id: 2148385,
      url: 'https://www.pivotaltracker.com/story/show/151356728',
      project_id: 2113469,
      owner_ids: [2148385],
      labels: [
        {
          id: 19539797,
          project_id: 2113469,
          kind: 'label',
          name: 'critical',
          created_at: '2017-12-17T17:06:44Z',
          updated_at: '2017-12-17T17:06:44Z',
        },
        {
          id: 19539795,
          project_id: 2113469,
          kind: 'label',
          name: 'qa by dev',
          created_at: '2017-12-17T17:06:33Z',
          updated_at: '2017-12-17T17:06:33Z',
        },
        {
          id: 19539796,
          project_id: 2113469,
          kind: 'label',
          name: 'test',
          created_at: '2017-12-17T17:06:35Z',
          updated_at: '2017-12-17T17:06:35Z',
        },
      ],
      owned_by_id: 2148385,
    },
  }
}
