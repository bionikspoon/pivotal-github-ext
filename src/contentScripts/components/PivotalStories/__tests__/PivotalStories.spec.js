import React from 'react'
import { shallow } from 'enzyme'
import PivotalStories from '../PivotalStories'

describe('components/PivotalStories', () => {
  test('it renders correctly', () => {
    const props = {
      storyIds: ['1234', '5678'],
    }
    const wrapper = shallow(<PivotalStories {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
