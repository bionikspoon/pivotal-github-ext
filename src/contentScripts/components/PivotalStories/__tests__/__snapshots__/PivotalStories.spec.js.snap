// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PivotalStories it renders correctly 1`] = `
<div
  className="PivotalStories__container"
>
  <withStateHandlers(lifecycle(mapProps(mapProps(branch(withProps(withStateHandlers(lifecycle(mapProps(mapProps(mapProps(branch(branch(branch(PivotalStory))))))))))))))
    key="1234"
    storyId="1234"
  />
  <withStateHandlers(lifecycle(mapProps(mapProps(branch(withProps(withStateHandlers(lifecycle(mapProps(mapProps(mapProps(branch(branch(branch(PivotalStory))))))))))))))
    key="5678"
    storyId="5678"
  />
</div>
`;
